WordPress/Classes/Services/PostService+RefreshStatus.swift

```diff
@@ -9,22 +9,24 @@ extension PostService {
     ///
     func refreshPostStatus(onCompletion: (() -> Void)? = nil, onError: ((Error) -> Void)? = nil) {
         self.managedObjectContext.perform {
-            let fetch = NSFetchRequest<Post>(entityName: Post.classNameWithoutNamespaces())
+            let request = NSBatchUpdateRequest(entityName: Post.classNameWithoutNamespaces())
             let pushingPredicate = NSPredicate(format: "remoteStatusNumber = %@", NSNumber(value: AbstractPostRemoteStatus.pushing.rawValue))
             let processingPredicate = NSPredicate(format: "remoteStatusNumber = %@", NSNumber(value: AbstractPostRemoteStatus.pushingMedia.rawValue))
-            fetch.predicate = NSCompoundPredicate(orPredicateWithSubpredicates: [pushingPredicate, processingPredicate])
+            let pushingOrProcessingPredicate = NSCompoundPredicate(orPredicateWithSubpredicates: [pushingPredicate, processingPredicate])
+            let notFailedPredicate = NSPredicate(format: "remoteStatusNumber != %@ AND NOT (postID != nil AND postID > 0)", NSNumber(value: AbstractPostRemoteStatus.failed.rawValue))
```

Comment:
Why are we checking for NOT (postID != nil AND postID > 0) here ?

Reply:
this is a representation of the `post.hasRemote()` logic from `markAsFailedAndDraftIfNeeded` method